src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -78,13 +37,7 @@ class TotalEnergy(BaseEnergy):
     """
 
     # ? add a generic contributions quantity to PhysicalProperty
-    contributions = SubSection(sub_section=EnergyContribution.m_def, repeats=True)
-
-    def __init__(
-        self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs
-    ) -> None:
-        super().__init__(m_def, m_context, **kwargs)
-        self.name = self.m_def.name
+    contributions = SubSection(sub_section=SectionProxy('TotalEnergy'), repeats=True)
```

Comment:
- Shouldn't the contributions be BaseEnergy's?
- How can you directly link the contribution to the Hamiltonian term with this implementation?

Reply:
I'm sorry, I don't understand what you mean here, perhaps we can discuss in our meeting